lib/wallet/http.js

```diff
@@ -2164,6 +1751,40 @@ class HTTP extends Server {
       return null;
     });
   }
+
+  /**
+   * Registers a new http handler at provided path
+   * the handler method is passed regular req, res along
+   * with a context object that holds vallet specific
+   * objects that might be used to extend wallet plugin
+   * functionality
+   *
+   * @param {*} httpMethod string value of one of
+   * 'post', 'get', 'put', 'patch', 'delete'
+   * @param {*} path location handler is listening
+   * @param {*} handler http handler that will handle requests
+   * arriving to `path` registered
+   * @param {Object} context context
+   * @returns {undefined}
+   */
+  registerRoute(httpMethod, path, handler, context) {
+    const validHttpMethods = ['post', 'get', 'put', 'patch', 'del'];
+    assert(httpMethod && typeof httpMethod === 'string',
+    'httpMethod parameter must be provided and it must be a string');
+    assert(validHttpMethods.includes(httpMethod.toLowerCase()),
+    'httpMethod must be one of ' + validHttpMethods.join(','));
+
+    // update context
+    context.TransactionOptions = TransactionOptions;
```

Comment:
Seems like some of these can be imported directly from the plugin, what is the benefit to passing these in here? Only `this.network` seems like actual state that belongs in the context to me, what do you think

Reply:
TransactionOptions is declared within wallet http.js and it's not exported. I also agree if we can directly import from the hsd, we should do that way. If there are any i missed please let me know ...

```
class TransactionOptions {
  /**
   * TransactionOptions
   * @alias module:http.TransactionOptions
   * @constructor
   * @param {Validator} valid
   */

  constructor(valid) {
    if (valid)
      return this.fromValidator(valid);
  }

  /**
   * Inject properties from Validator.
   * @private
   * @param {Validator} valid
   * @returns {TransactionOptions}
   */

  fromValidator(valid) {
    assert(valid);

    this.rate = valid.u64('rate');
    this.maxFee = valid.u64('maxFee');
    this.selection = valid.str('selection');
    this.smart = valid.bool('smart');
    this.account = valid.str('account');
    this.locktime = valid.u64('locktime');
    this.sort = valid.bool('sort');
    this.subtractFee = valid.bool('subtractFee');
    this.subtractIndex = valid.i32('subtractIndex');
    this.depth = valid.u32(['confirmations', 'depth']);
    this.paths = valid.bool('paths');
    this.outputs = [];

    if (valid.has('outputs')) {
      const outputs = valid.array('outputs');

      for (const output of outputs) {
        const valid = new Validator(output);

        let addr = valid.str('address');

        if (addr)
          addr = Address.fromString(addr, this.network);

        let covenant = valid.obj('covenant');

        if (covenant)
          covenant = Covenant.fromJSON(covenant);

        this.outputs.push({
          value: valid.u64('value'),
          address: addr,
          covenant: covenant
        });
      }
    }

    return this;
  }

  /*
   * Instantiate transaction options
   * from Validator.
   * @param {Validator} valid
   * @returns {TransactionOptions}
   */

  static fromValidator(valid) {
    return new this().fromValidator(valid);
  }
}
```